src/Sylius/Component/Taxonomy/Generator/TaxonSlugGenerator.php

```diff
@@ -52,6 +52,6 @@ public function generate($name, $parentId = null)
         $parent = $this->taxonRepository->find($parentId);
         Assert::notNull($parent, sprintf('There is no parent taxon with id %d.', $parentId));
 
-        return $parent->getSlug().self::SLUG_SEPARATOR.$taxonSlug;
+        return $parent->getTranslation($locale)->getSlug().self::SLUG_SEPARATOR.$taxonSlug;
```

Comment:
I don't think the issue is here, I think I wrote the Behat test wrong, let me check